modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -251,6 +254,7 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
       );
     }
     this._network = network;
+    this._amountType = amountType;
```

Comment:
since this is fully dependent on network let's do this instead

```suggestion
    this.amountType = utxolib.getMainnet(network) === utxolib.networks.dogecoin ? 'bigint' : 'number';
```

also the leading underscore is not needed here

Reply:
oh, I thought the leading underscore was meant for `private` variables

regarding the network dependency, I think the implementation is cleaner with the `amountType` being determined by each coin's individual override code, i.e. see `modules/sdk-coin-doge/src/doge.ts`

that way, there isn't any mention of `doge` within `abstract-utxo`

Reply:
ok fair enough

maybe we should add `utxolib.getAmountType(network): 'number' | 'bigint'`  at some point

Reply:
also I ended up changing it to `public` anyways, removed underscore